doc/developer/design/20231117_copy_to_s3.md

```diff
@@ -0,0 +1,305 @@
+# AWS Connection
+
+## The Problem
+
+Github issue:  [Issue 7256](https://github.com/MaterializeInc/materialize/issues/7256)
+
+As an operational data warehouse, Materialize sits upstream of a traditional data warehouse
+for our customers. The data brought in to Materialize will serve not only operational workloads,
+but ultimately analytical ones. This means our customers can benefit from a cheap and
+user-friendly method to share their data or send it on to the tools purpose-built for
+their analytical workloads while reducing the need to duplicate transformation logic
+across operational and analytical data warehouses. A full copy of data from Materialize
+into S3 is a simple and intuitive way to enable this for our customers.
+
+## Goals
+* Allow batch exports of data in Materialize to S3 in a format that can be easily ingested into analytical data warehouses.
+* Provide basic internal and external observability into the frequency, reliability, and size of the batch exports.
+* Tightly constrain scope so that the initial feature can reach public preview in early Q1 next year.
+
+## Non-goals
+* Supporting continuous or scheduled exports.
+* Integrating directly with batch data warehouses, without writing to S3.
+
+## Solution Proposal
+
+### SQL
+In materialize we already have some support for `COPY`
+commands. We should expand upon them, keeping them as close to
+the [postgres syntax](https://www.postgresql.org/docs/current/sql-copy.html).
+
+The SQL would look like the following.
+
+```sql
+COPY <mv_name or select_query> -- mv or table name or a valid SELECT query
```

Comment:
Is there a reason why this is limited to MVs or tables? Why can't we copy the contents of, e.g., a view or a source?

Reply:
Ah yes. Technically it's any relation + SELECT queries. Updating.